### ISSUE TITLE
Button Styles

### ISSUE THREAD

**Issue by sarah0arnold** [aemsites/clarkcountynv#1353]
**Component block:** button (and default state of link?)

**Design:** https://www.figma.com/design/T2bgro0OiuLqxvm1uLhJZ2/Clark-County-NV?node-id=93-224&p=f&t=7ZXrRUQI9M74s5dW-0

As a site visitor, I want to clearly see an actionable button so that I know where to click to view more information or take an action. 

**Requirements:** 

Web liaison should be able to configure the button text and destination link for the button
Add style variations to the button for the primary, secondary outline, and with icon. Each variation should include hover, focus, active (on click), and disabled states as defined by the brand guidelines/designs

https://www.aem.live/developer/block-collection/buttons

Primary style: 

- 600 primary ramp color with white bold button label text
- 300 primary ramp color on hover and focus
- 900 primary ramp color on click/press
- 400 neutral ramp color when disabled

Secondary outline style: 

- 600 primary ramp color outline and text color
- light opacity color fill on hover
- light opacity outline on focus
- 900 primary ramp color outline and text color with light opacity fill on press
- 400 neutral ramp color outline and text color when disabled

Need classes (example: [Explore More [[class: button section]]](https://main--clarkcountynv--aemsites.aem.page/government/elected_officials/county_treasurer/)) for styling buttons within another block
Buttons should be accessible via keyboard navigation
Buttons should be accessible and legible via screen reader
Default state: hyperlink text - author has to intentionally add bold or italics to create the button display

**Acceptance Criteria:** 
1. Given I am accessing the Sidekick Library, when I view the component list, then I can add a button block to my word document.
2. Given I am authoring a word document for a page, when I configure a button, then I am able to display an actionable button with a link and button text.
3. Given I am authoring a button block, when I edit the block, then I am able to specify the button style.
4. Given I have authored a button with a specific style, when I preview the page, then the button renders with the style I selected. 
5. Given I am viewing a web page with a primary button, when the page renders, then it displays with the 600 primary color and white label text. 
6. Given I am viewing a primary styled button, when I hover the button, then the button color changes to the 300 primary ramp color.
7. Given I am viewing a primary styled button, when I click the button, then the button color changes to the 900 primary ramp color on click.
8. Given I am using keyboard navigation, when I tab to a primary color button, then the button changes to 300 primary color (on focus). 
9. Given I am viewing a page with a primary styled button, when the button is disabled for any reason, then it displays with the 400 neutral color ramp color and is not clickable.
10. Given I am viewing a web page with a secondary outline button, when the page renders, then it displays with the 600 primary color outline and text color. 
11. Given I am viewing a secondary outline styled button, when I hover the button, then the button includes a light opacity fill.
12. Given I am viewing a primary styled button, when I click the button, then the button color changes to the 900 primary ramp color outline and text color on click.
13. Given I am using keyboard navigation, when I tab to a primary color button, then the button changes to light opacity outline (on focus). 
14. Given I am viewing a page with a primary styled button, when the button is disabled for any reason, then it displays with the 400 neutral color ramp color outline and text color and is not clickable.
15. Given I am authoring a button within another block (columns), when I apply a specific class, then I am able to display a styled button within other component blocks. 
16. Given I am viewing a button with an icon, when the button renders, then the icon displays to the left of the button text in the same color as the button text. 
17. Given I am navigating a web page with a button, when I use my keyboard to navigate through the page, then I am able to get to and press the button to navigate to the destination link.
18. Given I am using a screen reader, when I navigate to a page with a button, then the screen reader is able to announce the button in logical order on the page and I can understand what component is being used.

**Comment by beccasuchanek** [comment]
@aaronstewart-dev This ticket needs review for icon use case acceptance criteria, see details below:

Test Page URL: https://develop--clarkcountynv--aemsites.aem.page/drafts/bsuchanek/tickets/button-styles

16. ❌Given I am viewing a button with an icon, when the button renders, then the icon displays to the left of the button text in the same color as the button text. **On the secondary button with an icon, the icon does not appear primary color**

<img width="1489" height="824" alt="Image" src="https://github.com/user-attachments/assets/7fde8845-5a7f-47c3-b762-2dc0193b904e" />

**Status of ACs**
1. ✅ Given I am accessing the Sidekick Library, when I view the component list, then I can add a button block to my word document.
2. ✅ Given I am authoring a word document for a page, when I configure a button, then I am able to display an actionable button with a link and button text.
3. ✅ Given I am authoring a button block, when I edit the block, then I am able to specify the button style.
4. ✅ Given I have authored a button with a specific style, when I preview the page, then the button renders with the style I selected.
5. ✅ Given I am viewing a web page with a primary button, when the page renders, then it displays with the 600 primary color and white label text.
6. ✅ Given I am viewing a primary styled button, when I hover the button, then the button color changes to the 300 primary ramp color.
7. ✅ Given I am viewing a primary styled button, when I click the button, then the button color changes to the 900 primary ramp color on click.
8. ✅ Given I am using keyboard navigation, when I tab to a primary color button, then the button changes to 300 primary color (on focus).
9. ✅ Given I am viewing a page with a primary styled button, when the button is disabled for any reason, then it displays with the 400 neutral color ramp color and is not clickable.
10. ✅ Given I am viewing a web page with a secondary outline button, when the page renders, then it displays with the 600 primary color outline and text color.
11. ✅ Given I am viewing a secondary outline styled button, when I hover the button, then the button includes a light opacity fill.
12. ✅ Given I am viewing a primary styled button, when I click the button, then the button color changes to the 900 primary ramp color outline and text color on click.
13. ✅ Given I am using keyboard navigation, when I tab to a primary color button, then the button changes to light opacity outline (on focus).
14. ⚠️ Given I am viewing a page with a primary styled button, when the button is disabled for any reason, then it displays with the 400 neutral color ramp color outline and text color and is not clickable. **Design Attention Needed for this update, see ticket here https://github.com/aemsites/clarkcountynv/issues/1432**
15. ✅ Given I am authoring a button within another block (columns), when I apply a specific class, then I am able to display a styled button within other component blocks.
16. ❌Given I am viewing a button with an icon, when the button renders, then the icon displays to the left of the button text in the same color as the button text.
17. ✅ Given I am navigating a web page with a button, when I use my keyboard to navigate through the page, then I am able to get to and press the button to navigate to the destination link.
18. ✅ Given I am using a screen reader, when I navigate to a page with a button, then the screen reader is able to announce the button in logical order on the page and I can understand what component is being used.

Note: I will add the testing proof after the icon issue is fixed.
